File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/issue/GitIssueOpener.java
package org.ifinalframework.jetbrains.plugins.aio.issue;


import com.intellij.psi.PsiElement;
import com.intellij.psi.javadoc.PsiDocTag;
import org.ifinalframework.jetbrains.plugins.aio.api.idea.util.DocHelper;
import org.ifinalframework.jetbrains.plugins.aio.application.ElementHandler;
import org.ifinalframework.jetbrains.plugins.aio.browser.BrowserOpener;
import org.ifinalframework.jetbrains.plugins.aio.git.GitHelper;
import org.springframework.stereotype.Component;

import javax.annotation.Resource;

/**
 * 默认的Issue（浏览器）打开器
 *
 * @author iimik
 * @since 0.0.1
 * @see JiraIssueOpener
 **/
@Component
public class GitIssueOpener implements ElementHandler {
    /**
     * 浏览器打开工具
     */
    @Resource
    private BrowserOpener browserOpener;
    @Resource
    private DocHelper docHelper;
    @Resource
    private GitHelper gitHelper;

    @Override
    public void handle(PsiElement element) {

        if (!(element instanceof PsiDocTag)) {
            return;
        }

        PsiDocTag docTag = (PsiDocTag) element;
        final String tagName = docTag.getName();

        if (!IssueType.issue.name().equalsIgnoreCase(tagName)) {
            return;
        }

        final String value = docHelper.getDocTagValue(docTag);
        final String issuesUrl = gitHelper.getIssuesUrl(element, value);

        browserOpener.open(issuesUrl);


    }
}

File: idea-plugin/src/main/kotlin/org/ifinalframework/plugins/jetbrains/aio/issue/GitIssueOpener.kt
package org.ifinalframework.plugins.jetbrains.aio.issue;

import com.intellij.psi.PsiElement
import org.ifinalframework.plugins.jetbrains.aio.service.DocService
import org.springframework.stereotype.Component
import javax.annotation.Resource


/**
 * GitIssueOpener
 *
 * @author iimik
 * @since 0.0.1
 * @see JiraIssueOpener
 **/
@Component
class GitIssueOpener : IssueOpener {

    @Resource
    private lateinit var element: PsiElement

    /**
     * 浏览器打开工具
     */
    @Resource
    private lateinit var browserOpener: org.ifinalframework.plugins.jetbrains.aio.browser.BrowserOpener

    @Resource
    private lateinit var docService: DocService

    override fun supported(issueType: IssueType): Boolean {
        return IssueType.ISSUE == issueType
    }

    @Resource
    private lateinit var gitHelper: org.ifinalframework.plugins.jetbrains.aio.git.GitHelper


    override fun open(issue: Issue) {
        val issuesUrl = gitHelper.getIssuesUrl(element, issue.code)
        browserOpener!!.open(issuesUrl)
    }

}